test/DatabaseTest.cc

```diff
@@ -208,1046 +251,174 @@ TEST_CASE_METHOD(DatabaseTest, "Database w/o config") {
+    REQUIRE(correctkeydb);
+    CBLDatabase_Release(correctkeydb);
     
-    CHECK(CBLDatabase_PurgeDocument(db, doc, &error));
-    CBLDocument_Release(doc);
-    CHECK(!CBLDatabase_GetDocument(db, "doc1"_sl, &error));
+    // No key:
+    {
+        ExpectingExceptions x;
+        CBLDatabase *nokeydb = CBLDatabase_Open("encdb"_sl, nullptr, &error);
+        REQUIRE(nokeydb == nullptr);
+        CHECK(error.domain == kCBLDomain);
+        CHECK(error.code == kCBLErrorNotADatabaseFile);
+    }
     
-    CHECK(CBLDatabase_PurgeDocumentByID(db, "doc2"_sl, &error));
-    CHECK(!CBLDatabase_GetDocument(db, "doc2"_sl, &error));
+    // Wrong key:
+    {
+        ExpectingExceptions x;
+        CBLEncryptionKey key2;
+        
+        if (useSHA256Key) {
+            CBLEncryptionKey_FromPassword(&key2, "wrongpassword"_sl);
+        } else {
+            CBLEncryptionKey_FromPasswordOld(&key2, "wrongpassword"_sl);
+        }
+        
+        CBLDatabaseConfiguration config2 = {nullslice, key2};
+        CBLDatabase *wrongkeydb = CBLDatabase_Open("encdb"_sl, &config2, &error);
+        REQUIRE(wrongkeydb == nullptr);
+        CHECK(error.domain == kCBLDomain);
+        CHECK(error.code == kCBLErrorNotADatabaseFile);
+    }
+
+    CHECK(CBLDatabase_Delete(defaultdb, &error));
+    CBLDatabase_Release(defaultdb);
+    CHECK(!CBL_DatabaseExists("encdb"_sl, nullslice));
 }
 
+#endif
+
+#pragma mark - Save Document:
 
-TEST_CASE_METHOD(DatabaseTest, "Legacy - Purge Already Purged Document") {
-    createDocument(db, "doc1", "foo", "bar");
+TEST_CASE_METHOD(DatabaseTest, "Save Document into Different DB Instance") {
+    CBLDocument* doc = CBLDocument_CreateWithID("foo"_sl);
+    FLMutableDict props = CBLDocument_MutableProperties(doc);
+    FLMutableDict_SetString(props, "greeting"_sl, "Howdy!"_sl);
     
     CBLError error;
-    const CBLDocument* doc = CBLDatabase_GetDocument(db, "doc1"_sl, &error);
-    REQUIRE(doc);
-    
-    CHECK(CBLDatabase_PurgeDocument(db, doc, &error));
-    CHECK(!CBLDatabase_GetDocument(db, "doc1"_sl, &error));
-    
-    CHECK(!CBLDatabase_PurgeDocument(db, doc, &error));
-    CBLDocument_Release(doc);
-    CHECK(error.domain == kCBLDomain);
-    CHECK(error.code == kCBLErrorNotFound);
+    CHECK(CBLCollection_SaveDocument(col, doc, &error));
     
-    error = {};
-    CHECK(!CBLDatabase_PurgeDocumentByID(db, "doc1"_sl, &error));
+    ExpectingExceptions x;
+    CHECK(!CBLCollection_SaveDocument(otherDBDefaultCol, doc, &error));
     CHECK(error.domain == kCBLDomain);
-    CHECK(error.code == kCBLErrorNotFound);
+    CHECK(error.code == kCBLErrorInvalidParameter);
+    CBLDocument_Release(doc);
 }
 
+#pragma mark - Delete Document:
 
-TEST_CASE_METHOD(DatabaseTest, "Legacy - Purge Document from Different DB") {
-    createDocument(db, "doc1", "foo", "bar");
-    
+TEST_CASE_METHOD(DatabaseTest, "Delete Document from Different DB Instance") {
```

Comment:
I think this test can be deleted as we have the same test but with collection in DocumentTest. Please double check.

Reply:
Operation is performed on 2 different dbs default collection, rather than 2 collections within the same db. It is pretty much the same and if is still considered redundant then will remove.